code-studio/build-commands.js

```diff
@@ -46,20 +46,57 @@ exports.browserify = function (config) {
 
   var extension = (shouldMinify ? '.min.js' : '.js');
 
+  // The React library checks the NODE_ENV variable in several places to decide
+  // whether to enable special debugging features.  By setting this environment
+  // variable, envify will do an inline replacement anywhere `process.env.NODE_ENV`
+  // is used (envify is implicitly part of the browserify step, since we depend
+  // on 'react') and then uglify will perform dead code removal that strips all
+  // of those debug paths from our minified output.
+  var customEnvironment = (shouldMinify ? 'NODE_ENV=production' : '');
+
   var command = (shouldWatch ? 'watchify -v' : 'browserify') +
     (shouldMinify ? '' : ' --debug');
+  
+  var babelifyStep = '-t [ babelify --compact=false --sourceMap --sourceMapRelative="$PWD" ]';
+
+  // Uglify shrinks our output down as small as possible.
+  // --compress performs lots of optimizations including removal of dead code.
+  // warnings=false disables compressor warnings (which don't indicate actual problems, for us)
+  // --mangle replaces variable names with short representations
+  var uglifyCommand = 'uglifyjs --compress warnings=false --mangle';
```

Comment:
@wjordan pointed out that I wasn't doing the most important uglify steps - compression and mangling.

Fixing this was important for having a "production" React.  When building with React it inlines the NODE_ENV environment variable into the code (via `envify` which is one of its dependencies) so a bunch of its debug stuff ends up behind `if ("production" !== "production")` checks.  It expects that you'll use an uglifier that removes dead code to actually avoid shipping this stuff.

We disable compressor warnings because by default uglify spits out a warning every time it strips any dead code - which is a lot, thanks to React's approach to dev/prod builds.  We don't need that build noise.  If someone wants to go back and investigate these warnings in detail, they're welcome to, but they're hard to track down anyway right now since we're not passing a sourcemap into this uglify step (yet).